hydra-node/src/Hydra/API/Server.hs

```diff
@@ -139,8 +140,9 @@ withAPIServer ::
   PersistenceIncremental (TimedServerOutput tx) IO ->
   Tracer IO APIServerLog ->
   Chain tx IO ->
+  ProtocolParameters ->
   ServerComponent tx IO ()
-withAPIServer host port party PersistenceIncremental{loadAll, append} tracer chain callback action = do
+withAPIServer host port party PersistenceIncremental{loadAll, append} tracer chain pparams callback action = do
```

Comment:
I can second this. I also see the the risk of `Hydra.API` becoming very cardano-specific.

However, I'm not sure if there is an easy way to avoid this?

Reply:
I remember you had the same comment when we were working on this piece of code together @abailly-iohk The thing is, this function is called only from one place and I am not sure if using some custom data type (or ReaderT) would improve the situation.

Reply:
I don't see an obvious way to improve the situation, I would need to spend some time together to work over it.